opennlp-tools/src/main/java/opennlp/tools/sentiment/SentimentME.java

```diff
@@ -0,0 +1,280 @@
+import opennlp.tools.util.Span;
+import opennlp.tools.util.TrainingParameters;
+import opennlp.tools.util.featuregen.AdaptiveFeatureGenerator;
+import opennlp.tools.util.featuregen.AdditionalContextFeatureGenerator;
+
+/**
+ * Class for creating a maximum-entropy-based Sentiment Analysis model.
+ */
+public class SentimentME {
+
+  public static final String OTHER = "other";
+  public static final String START = "start";
+  public static final String CONTINUE = "cont";
+  public static final int DEFAULT_BEAM_SIZE = 3;
+
+  private static String[][] EMPTY = new String[0][0];
+
+  protected SentimentContextGenerator contextGenerator;
+  private AdditionalContextFeatureGenerator additionalContextFeatureGenerator =
+      new AdditionalContextFeatureGenerator();
+  private Sequence bestSequence;
+  protected SequenceClassificationModel<String> model;
+  private SequenceValidator<String> sequenceValidator;
+  private SentimentFactory factory;
+  private MaxentModel maxentModel;
+  private SequenceCodec<String> seqCodec = new BioCodec();
+  private AdaptiveFeatureGenerator[] featureGenerators;
+
+  /**
+   * Constructor, initialises
+   *
+   * @param sentModel
+   *          sentiment analysis model
+   */
+  public SentimentME(SentimentModel sentModel) {
+
+    this.model = sentModel.getSentimentModel();
+    maxentModel = sentModel.getMaxentModel();
+
+    factory = sentModel.getFactory();
+
+    contextGenerator = factory.createContextGenerator();
+  }
+
+  /**
+   * Trains a Sentiment Analysis model.
+   *
+   * @param languageCode
+   *          the code for the language of the text, e.g. "en"
+   * @param samples
+   *          the sentiment samples to be used
+   * @param trainParams
+   *          parameters for training
+   * @param factory
+   *          a Sentiment Analysis factory
+   * @return a Sentiment Analysis model
+   */
+  public static SentimentModel train(String languageCode,
+      ObjectStream<SentimentSample> samples, TrainingParameters trainParams,
+      SentimentFactory factory) throws IOException {
+
+    Map<String, String> entries = new HashMap<String, String>();
+
+    MaxentModel sentimentModel = null;
+
+    ObjectStream<Event> eventStream = new SentimentEventStream(samples,
+        factory.createContextGenerator());
+
+    EventTrainer trainer = TrainerFactory
+        .getEventTrainer(trainParams, entries);
+    sentimentModel = trainer.train(eventStream);
+
+    Map<String, String> manifestInfoEntries = new HashMap<String, String>();
+
+    return new SentimentModel(languageCode, sentimentModel, manifestInfoEntries,
+        factory);
+
+  }
+
+  /**
+   * Makes a sentiment prediction
+   *
+   * @param sentence
+   *          the text to be analysed for its sentiment
+   * @return the predicted sentiment
+   */
+  public String predict(String sentence) {
+    String[] tokens = factory.getTokenizer().tokenize(sentence);
+
+    return predict(tokens);
+  }
+
+  public String predict(String[] tokens) {
```

Comment:
The naming of the methods to detect the sentiment have to be improves, it is hard to understand which one should be used. There is predict, predict2 and find.